lms/djangoapps/completion/tests/test_models.py

```diff
@@ -142,3 +142,52 @@ def test_cannot_call_submit_completion(self):
                 completion=0.9,
             )
         self.assertEqual(models.BlockCompletion.objects.count(), 1)
+
+
+class SubmitBatchCompletionTestCase(TestCase):
+    """
+    Test that BlockCompletion.objects.submit_batch_completion has the desired
+    semantics.
+    """
+
+    def setUp(self):
+        super(SubmitBatchCompletionTestCase, self).setUp()
+        _overrider = waffle.waffle().override(waffle.ENABLE_COMPLETION_TRACKING, True)
+        _overrider.__enter__()
+        self.addCleanup(_overrider.__exit__, None, None, None)
+
+        blocks = [
+            (UsageKey.from_string('block-v1:edx+test+run+type@video+block@doggos'), 1.0),
+        ]
+        self.course_key = 'course-v1:edx+test+run'
+        self.course_key_obj = CourseKey.from_string(self.course_key)
+        self.blocks = blocks
+        self.user = UserFactory()
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course_key)
+
+    def test_submit_batch_completion(self):
+        models.BlockCompletion.objects.submit_batch_completion(self.user, self.course_key_obj, self.blocks)
+        self.assertEqual(models.BlockCompletion.objects.count(), 1)
+        self.assertEqual(models.BlockCompletion.objects.last().completion, 1.0)
+
+    def test_submit_batch_completion_without_waffle(self):
+        with waffle.waffle().override(waffle.ENABLE_COMPLETION_TRACKING, False):
+            with self.assertRaises(RuntimeError):
+                models.BlockCompletion.objects.submit_batch_completion(self.user, self.course_key_obj, self.blocks)
+
+    def test_submit_batch_completion_with_same_block_new_completion_value(self):
+        blocks = [
```

Comment:
This is the same block_id as the one in `self.blocks`.  Instead of carrying that around, how about we initialized just a block_id in the setUp, and then use that as needed for different tests?  So in your `setUp()` you could have:
```python
self.block_id = UsageKey.from_string('block-v1:edx+test+run+type@video+block@doggos'
```
and then in this function, you could pass `[(self.block_id, 0.0)]` to `submit_batch_completion()`, and similarly for the tests above.